hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/BackupMasterObserver.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.backup;
+
+import java.io.IOException;
+import java.util.Optional;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseInterfaceAudience;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.backup.impl.BackupManager;
+import org.apache.hadoop.hbase.backup.impl.BackupSystemTable;
+import org.apache.hadoop.hbase.coprocessor.MasterCoprocessor;
+import org.apache.hadoop.hbase.coprocessor.MasterCoprocessorEnvironment;
+import org.apache.hadoop.hbase.coprocessor.MasterObserver;
+import org.apache.hadoop.hbase.coprocessor.ObserverContext;
+import org.apache.yetus.audience.InterfaceAudience;
+
+@InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.CONFIG)
+public class BackupMasterObserver implements MasterObserver, MasterCoprocessor {
+
+  @Override
+  public void preTruncateTable(ObserverContext<MasterCoprocessorEnvironment> ctx,
+    TableName tableName) throws IOException {
+    disallowIncrementalBackups(ctx.getEnvironment(), tableName);
+  }
+
+  @Override
+  public void postTruncateTable(ObserverContext<MasterCoprocessorEnvironment> ctx,
+    TableName tableName) throws IOException {
+    disallowIncrementalBackups(ctx.getEnvironment(), tableName);
+  }
+
+  @Override
+  public Optional<MasterObserver> getMasterObserver() {
+    return Optional.of(this);
+  }
+
+  private static void disallowIncrementalBackups(MasterCoprocessorEnvironment env,
+    TableName tableName) throws IOException {
+    Configuration conf = env.getConfiguration();
+    if (
+      tableName.equals(BackupSystemTable.getTableName(conf)) || !BackupManager.isBackupEnabled(conf)
+    ) {
+      return;
+    }
+
+    BackupSystemTable table = new BackupSystemTable(env.getConnection());
+    try {
+      table.startBackupExclusiveOperation();
```

Comment:
Does it make more sense to include lock acquisition/release inside the `disallowFurtherIncremetnals` method? That's not the pattern that we follow now, but I wonder if we should to make sure each caller isn't required to take out the lock